std/datetime/date.d

```diff
@@ -2861,23 +2880,42 @@ public:
 
     /++
         Converts this $(LREF DateTime) to a string with the format
-        YYYY-MM-DDTHH:MM:SS.
+        `YYYY-MM-DDTHH:MM:SS`. If `writer` is set, the resulting
+        string will be written directly to it.
+
+        Params:
+            writer = A `char` accepting
+            $(REF_ALTTEXT output range, isOutputRange, std, range, primitives)
+        Returns:
+            A `string` when not using an output range; `void` otherwise.
       +/
     string toISOExtString() const @safe pure nothrow
     {
-        import std.format : format;
+        import std.array : appender;
+        auto w = appender!string();
+        w.reserve(20);
+        toISOExtString(w);
+        return w.data;
+    }
+
+    /// ditto
+    void toISOExtString(W)(ref W writer) const
+    if (isOutputRange!(W, char))
+    {
+        import std.format : formattedWrite;
         try
         {
-            return format!("%sT%02d:%02d:%02d")(
-                _date.toISOExtString(),
+            _date.toISOExtString(writer);
+            formattedWrite!("T%02d:%02d:%02d")(
+                writer,
                 _tod._hour,
                 _tod._minute,
                 _tod._second
             );
         }
         catch (Exception e)
         {
-            assert(0, "format() threw.");
+            assert(0, "formattedWrite() threw.");
```

Comment:
Does it make sense to have this try/catch assert when `writer` can be any arbitrary output range?